File: sprint4_app/ios/Runner/MethodChannelType.swift
import Foundation
import Flutter

enum MethodChannelType {
    case imageLabeling
    case platformIdentifier
    
    var propertyName: String {
        switch self {
            case .imageLabeling:
                "image_labeling"
            case .platformIdentifier:
                "platform_identifier"
        }
    }
    
    var methodName: String {
        switch self {
            case .imageLabeling:
                "labelImage"
            case .platformIdentifier:
                "identifyPlatform"
        }
    }
    
    private func channel(binaryMessenger: FlutterBinaryMessenger) -> FlutterMethodChannel {
        let channelName = "com.sprint4" + "/" + propertyName
        return FlutterMethodChannel(name: channelName, binaryMessenger: binaryMessenger)
    }
    
    func setMethodCallHandler(binaryMessenger: FlutterBinaryMessenger) {
        let channel = channel(binaryMessenger: binaryMessenger)
        
        channel.setMethodCallHandler {
            call,
            result in
            
            handleCall(call: call, result: result)
        }
    }
    
    private func handleCall(call: FlutterMethodCall, result: @escaping FlutterResult) {
        guard call.method == methodName else {
            result(FlutterMethodNotImplemented)
            return
        }
        
        switch self {
            case .imageLabeling:
                guard let args = call.arguments as? [String: Any],
                      let bytes = args["bytes"] as? FlutterStandardTypedData else {
                    
                    result(FlutterError(
                        code: "INVALID_ARGUMENTS",
                        message: "Missing image bytes",
                        details: nil
                    ))
                    
                    return
                }
            
                ImageLabelingService.labelImage(withBytes: bytes, result: result)
            default:
                return
        }
    }
}

File: sprint4_app/android/app/src/main/kotlin/com/example/sprint4_app/MethodChannelType.kt
package com.example.sprint4_app

import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

enum class MethodChannelType(
    val propertyName: String,
    val methodName: String
) {
    IMAGE_LABELING("image_labeling", "labelImage");

    private fun channel(binaryMessenger: BinaryMessenger): MethodChannel {
        val channelName = "com.sprint4/$propertyName"
        return MethodChannel(binaryMessenger, channelName)
    }

    fun setMethodCallHandler(binaryMessenger: BinaryMessenger) {
        val channel = channel(binaryMessenger)

        channel.setMethodCallHandler { call, result ->
            handleCall(call, result)
        }
    }

    private fun handleCall(call: MethodCall, result: MethodChannel.Result) {
        if (call.method != methodName) {
            result.notImplemented()
            return
        }

        when (this) {
            IMAGE_LABELING -> {
                val args = call.arguments as? Map<*, *>
                val bytes = args?.get("bytes") as? ByteArray

                if (bytes == null) {
                    result.error("INVALID_ARGUMENTS", "Missing image bytes", null)
                    return
                }

                ImageLabelingService.labelImage(bytes, result)
            }
        }
    }
}